.github/workflows/build-test.yml

```diff
@@ -1016,3 +1031,80 @@ jobs:
       # Alternative method for uploading
       # - name: Upload coverage to codecov
       #  run: bash <(curl -s https://codecov.io/bash) -f ./coverage/cobertura-coverage.xml -v
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    # This should happen in the release and insiders branch
+    if: github.repository == 'microsoft/vscode-jupyter' && (needs.pick_environment.outputs.release_channel == 'insider' || needs.pick_environment.outputs.release_channel == 'stable')
+    needs: [pick_environment, build-vsix, lint, ts_tests, smoke-tests]
+    environment:
+      name: ${{needs.pick_environment.outputs.release_channel}}
+    steps:
+      - name: dump_event
+        run: |
+          echo "Event_name is ${{ github.event_name }}"
+          echo "Ref is ${{ github.ref }}"
+          echo "Release Channel (Environment) is is ${{needs.pick_environment.outputs.release_channel}}"
+          echo "VSIX Name is ${{ needs.pick_environment.outputs.vsix_name }}"
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Use Python ${{env.PYTHON_VERSION}}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{env.PYTHON_VERSION}}
+
+      - name: Use Node ${{env.NODE_VERSION}}
+        uses: actions/setup-node@v2.1.5
+        with:
+          node-version: ${{env.NODE_VERSION}}
+
+      - name: Download VSIX
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{needs.pick_environment.outputs.vsix_name}}
+
+    #   - name: Publish
+    #     run: vsce publish --packagePath ${{needs.pick_environment.outputs.vsix_name}} --pat ${{secrets.VSCE_TOKEN}} --noVerify
```

Comment:
Once reviewed, i'll enable this, only way to find out is to push a VSIX for insiders and test.